### ISSUE TITLE
composer info --vendor

### ISSUE THREAD

**Issue by Soullivaneuh** [composer/composer#5212]
The goal of `--vendor` option is to easily get vendor specific packages from `composer info` command.

Some examples:

```bash
$ composer info --vendor=sonata-project
sonata-project/admin-bundle              dev-nexy d2fcca4    Symfony SonataAdminBundle
sonata-project/block-bundle              2.3.9               Symfony SonataBlockBundle
sonata-project/cache                     1.0.7               Cache library
sonata-project/core-bundle               2.3.10              Symfony SonataCoreBundle
sonata-project/doctrine-extensions       1.0.2               Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle dev-nexy 483576b    Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.1.10              Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.4.1               Lightweight Exporter library
sonata-project/intl-bundle               2.2.4               Symfony SonataIntlBundle
```

```bash
$ composer info --all --vendor=sonata-project
  sonata-project/admin-bundle
  sonata-project/admin-bundle-persistency-layer
  sonata-project/admin-search-bundle
  sonata-project/block-bundle
  sonata-project/cache
  sonata-project/cache-bundle
  sonata-project/classification-bundle
  sonata-project/comment-bundle
  sonata-project/composer-archive-creator
  sonata-project/core-bundle
  sonata-project/dashboard-bundle
  sonata-project/datagrid-bundle
  sonata-project/doctrine-extensions
  sonata-project/doctrine-mongodb-admin-bundle
  sonata-project/doctrine-orm-admin-bundle
  sonata-project/doctrine-phpcr-admin-bundle
  sonata-project/easy-extends-bundle
  sonata-project/ecommerce
  sonata-project/exporter
  sonata-project/formatter-bundle
  sonata-project/google-authenticator
  sonata-project/intl-bundle
  sonata-project/jquery-bundle
  sonata-project/markitup-bundle
  sonata-project/media-bundle
  sonata-project/news-bundle
  sonata-project/notification-bundle
  sonata-project/page-bundle
  sonata-project/propel-admin-bundle
  sonata-project/sandbox
  sonata-project/seo-bundle
  sonata-project/sonata-composer
  sonata-project/timeline-bundle
  sonata-project/translation-bundle
  sonata-project/user-bundle

installed:
  sonata-project/admin-bundle              dev-nexy d2fcca4    Symfony SonataAdminBundle
  sonata-project/block-bundle              2.3.9               Symfony SonataBlockBundle
  sonata-project/cache                     1.0.7               Cache library
  sonata-project/core-bundle               2.3.10              Symfony SonataCoreBundle
  sonata-project/doctrine-extensions       1.0.2               Doctrine2 behavioral extensions
  sonata-project/doctrine-orm-admin-bundle dev-nexy 483576b    Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
  sonata-project/easy-extends-bundle       2.1.10              Symfony SonataEasyExtendsBundle
  sonata-project/exporter                  1.4.1               Lightweight Exporter library
  sonata-project/intl-bundle               2.2.4               Symfony SonataIntlBundle
```

**Comment by curry684** [comment]
What is the added value over:
```sh
composer search sonata-project
composer show | grep "sonata-project"
```
?

**Comment by Soullivaneuh** [comment]
For `composer search sonata-project`:

AFAIK, `composer search` is for searching package around world wild web, not showing installed packages.

And BTW, `sonata-project` is not a vendor specific token in this way.

For `composer show | grep "sonata-project/"`:

First of all, check this command result:

```bash
$ composer show --all | grep "sonata-project/"
  sebastian-alfers-sonata-project/doctrine-mongodb-admin-bundle
  sonata-project/admin-bundle
  sonata-project/admin-bundle-persistency-layer
  sonata-project/admin-search-bundle
  sonata-project/block-bundle
  sonata-project/cache
  sonata-project/cache-bundle
  sonata-project/classification-bundle
  sonata-project/comment-bundle
  sonata-project/composer-archive-creator
  sonata-project/core-bundle
  sonata-project/dashboard-bundle
  sonata-project/datagrid-bundle
  # ...
```

`sebastian-alfers-sonata-project/doctrine-mongodb-admin-bundle` is not part of the `sonata-project` vendor, so it's not relevant.

The second issue is the usage of `grep`:

1. This would launch two process instead of one (yeah, process optimization matter!).
2. You method means you must have `grep` command or good equivalent. Everybody does not have this chance (Hi Windows users! :trollface:).

**Comment by stof** [comment]
@Soullivaneuh your argument about search being related to Packagist packages is irrelevant here. ``composer show --all`` is also related to all packages, not only to installed ones

**Comment by Soullivaneuh** [comment]
@stof I agree for `composer show --all` but what I'm saying it's `composer search` do **only** that.

Am I right?

**Comment by Seldaek** [comment]
Fixed in 21a9f21cc87a8173a8fa6c358b740b05fd9ad3c2 in a more valuable way IMO (and without yet-another-flag-nobody-will-remember:):

```
$ c show -a sym*ny/*
symfomany/canicule
symfomany/cinhetic
symfomany/laravel-mongo
symfomany/laravelcinema
symfomany/prez
symfony/asset
symfony/assetic-bundle
symfony/bootcamp-bundle
symfony/browser-kit
symfony/cache
```

```
$ c show seld/*
seld/cli-prompt 1.0.2 Allows you to prompt for user input on the command line, and optionally hide the characters...
seld/jsonlint   1.4.0 JSON Linter
seld/phar-utils 1.0.1 PHAR file format utilities, for when PHP phars you up
```
